### PR TITLE
fix(artifacts): cap azure reference artifact files at `max_objects`

### PR DESCRIPTION
**Pull request by amusipatla-wandb**
Description
-----------

This PR fixes how we handle `max_objects` when uploading Azure reference artifacts.
Previously, we didn't limit by `max_objects` but raised an error whenever we received more than `max_objects` files from the given path.
With this change, we instead stop adding entries once we reach `max_objects` entries.

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
Tested locally by uploading 10 files from a folder with 1mil+ files

### CODE REVIEW

File: wandb/sdk/artifacts/storage_handlers/azure_handler.py [hunk 1]
```diff
@@ -151,6 +148,13 @@ def store_path(
                         ),
                     )
                 )
+
+        if len(entries) > max_objects:
+            raise ValueError(
+                f"Exceeded {max_objects} objects tracked, pass max_objects to "
+                f"add_reference"
+            )
+
```
Comment (amusipatla-wandb) [inline wandb/sdk/artifacts/storage_handlers/azure_handler.py:157]: we shouldn't ever hit this given the check above, all other handlers also have checks even though they explicitly limit the number of objects they get, so I kept this in to match that pattern

**Review by estellazx — APPROVED** [review]

File: wandb/sdk/artifacts/storage_handlers/azure_handler.py [hunk 2]
```diff
@@ -136,10 +136,7 @@ def store_path(
             name_starts_with=f"{blob_name}/"
         ):
             if len(entries) >= max_objects:
-                raise ValueError(
-                    f"Exceeded {max_objects} objects tracked, pass max_objects to "
-                    f"add_reference"
-                )
+                break
```
Comment (dmitryduev) [inline wandb/sdk/artifacts/storage_handlers/azure_handler.py:139]: do you you want to print a warning to the user, perhaps?

**Review by dmitryduev — APPROVED** [review]

File: wandb/sdk/artifacts/storage_handlers/azure_handler.py (hunk 1 above)
Comment (amusipatla-wandb) [reply wandb/sdk/artifacts/storage_handlers/azure_handler.py]: just removed this anyway, we should never run into this

File: wandb/sdk/artifacts/storage_handlers/azure_handler.py (hunk 2 above)
Comment (amusipatla-wandb) [reply wandb/sdk/artifacts/storage_handlers/azure_handler.py:143]: added this!

**Pull request merged**